pytorch_lightning/accelerators/registry.py

```diff
@@ -0,0 +1,78 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import importlib
+from inspect import getmembers, isclass
+from pathlib import Path
+from typing import Any, List, Optional
+
+from pytorch_lightning.accelerators.accelerator import Accelerator
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+
+class _AcceleratorRegistry(dict):
+    def register(self, accelerator: Accelerator, name: Optional[str] = None, override: bool = False) -> Accelerator:
+        """Registers an accelerator mapped to a name.
+
+        Args:
+            accelerator: the accelerator to be mapped.
+            name: the name that identifies the provided accelerator.
+            override: Whether to override an existing key.
+        """
+        if name is None:
+            name = accelerator.name()
```

Comment:
strategy is using class variables instead of a property. Should we choose one of them for consistency?

Reply:
We need `name` to be an abstract method. We wish to make it mandatory for the users to define it, so they could pass the name to the `accelerator` flag to the Trainer.

```python
trainer = Trainer(accelerator="sota_accelerator", devices="auto")
```

Reply:
Also, the reason to make it a static method is to get the name without instantiating the Accelerator class

Reply:
@kaushikb11, I would still prefer class variables. They are also static, so you can access them without instantiation. And more important: They are the same for every instance of the class. Whereas the function approach could still contain some logic to change this dynamically. And I think, we could check that this has been set during the accelerator's init?

Reply:
@justusschock But the end goal is 

```python
accelerator_name = "custom_accelerator"

class TestAccelerator(Accelerator):
    @staticmethod
    def parse_devices(devices):
        return devices

    @staticmethod
    def get_parallel_devices(devices):
        return ["foo"] * devices

    @staticmethod
    def auto_device_count():
        return 3

    @staticmethod
    def is_available():
        return True

    @staticmethod
    def name():
        return accelerator_name

AcceleratorRegistry.register(TestAccelerator)

trainer = Trainer(accelerator="custom_accelerator", devices="auto")
assert isinstance(trainer.accelerator, TestAccelerator)
```

How would we enforce users to define the class variable if they wish to use their custom accelerator?

Reply:
Resolved offline, see https://github.com/PyTorchLightning/pytorch-lightning/pull/12180#discussion_r832509497